solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -48,6 +48,49 @@
   private final ZkNodeProps message;
   private final String collection; // maybe null
 
+  /** Shard / collection health state. */
+  public enum Health {
+    /** All replicas up, leader exists. */
+    GREEN,
+    /** Some replicas down, leader exists. */
+    YELLOW,
+    /** Many replicas down, leader exists. */
+    ORANGE,
+    /** No leader or most replicas down. */
+    RED;
+
+    public static final float ORANGE_LEVEL = 0.5f;
+    public static final float RED_LEVEL = 0.1f;
```

Comment:
RED_LEVEL could be 0.0, cause we require > RED_LEVEL. 
Then, if someone have 20 replicas, you'll be ORANGE instead of RED when 1 replica is alive.